tests/compiler_driver/precompile.mlir

```diff
@@ -0,0 +1,8 @@
+// RUN: iree-compile --compile-mode=precompile --iree-hal-target-backends=vmvx %s | iree-compile --iree-hal-target-backends=vmvx --compile-from=global-optimization --compile-to=stream - > %t && \
+// RUN: iree-compile --iree-hal-target-backends=vmvx --compile-to=stream %s | diff - %t
```

Comment:
This test only checks till stream, because as we convert to the dispatch binary, the binary starts differing. I'm not sure why that happens.

Reply:
Clever test :)

But I think we can go for something more smoke testy like the hal_executable.mlir. mainly want something covered for sanitizers and a sanity check.

Reply:
Just FileCheck and verify something basic with a check out check-not.

Reply:
(also, pretty sure diff like this will break on Windows)